python/revng/scripting/projects/cli_project.py

```diff
@@ -0,0 +1,106 @@
+        self._set_resume_dir(resume_dir)
+        self._set_revng_executable(revng_executable)
+        super().__init__()
+
+    def _load(self):
+        self._get_pipeline_description()
+        # TODO: We hardcode the path to the `model.yml`, fix when there is a
+        # command to get the model.
+        model_path = f"{self.resume_dir}/context/model.yml"
+        if not os.path.exists(model_path) and not os.path.isfile(model_path):
+            return
+        with open(model_path) as f:
+            self._set_model(Binary.deserialize(f))
+
+    def import_binary(self, input_binary_path: str):
+        self.input_binary = input_binary_path
+
+    def _get_artifact(self, artifact_name: str, targets_class: List[Function] = []) -> bytes | str:
+        targets = []
+        if targets_class:
+            kind = self._get_artifact_kind(artifact_name)
+            targets.extend([f"{tc.key()}:{kind}" for tc in targets_class])
+
+        result = self._get_artifact_internal(artifact_name, targets)
+        return self._get_result_mime(artifact_name, result)
+
+    def _get_artifact_internal(self, artifact_name: str, targets: List[str] = []) -> bytes | str:
+        args = [
+            self.revng_executable,
+            "artifact",
+            f"--resume={self.resume_dir}",
+            artifact_name,
+            self.input_binary,
+        ]
+        args.extend(targets)
+
+        return subprocess.run(args, capture_output=True, check=True).stdout
+
+    def analyze(self, analysis_name: str, targets={}, options={}):
+        # TODO: add `targets` when `revng`` allows it
+        args = [self.revng_executable, "analyze", f"--resume={self.resume_dir}", analysis_name]
+        if options:
+            for name, value in options.items():
+                args.append(f"--{analysis_name}-{name}={value}")
+        args.append(self.input_binary)
+
+        result = subprocess.run(args, capture_output=True, check=True).stdout
+        model = Binary.deserialize(result.decode("utf-8"))
+        self._set_model(model)
+
+    def analyses_list(self, analysis_name: str):
+        return self.analyze(analysis_name)
+
+    def commit(self):
+        diff = DiffSet.make(self.last_saved_model, self.model)
+        if len(diff.changes) == 0:
+            return
+
+        with NamedTemporaryFile(mode="w") as tmp_diff_file:
+            diff.serialize(tmp_diff_file)
+            tmp_diff_file.flush()
+
+            args = [
+                self.revng_executable,
+                "analyze",
+                f"--resume={self.resume_dir}",
+                "apply-diff",
+                "--apply-diff-global-name=model.yml",
+                f"--apply-diff-diff-content-path={tmp_diff_file.name}",
+                self.input_binary,
+            ]
+
+            model = subprocess.run(args, capture_output=True, check=True).stdout
+            self.last_saved_model = Binary.deserialize(model.decode("utf-8"))
+
+    def _get_pipeline_description(self):
```

Comment:
1. `get_*` methods should return something. I think this method should not exist at all. It should be inlined in its call site and not be part of the interface of the base class.
2. We seem to just ignore the fact the that we don't have `pipeline-description.yml`. I don't think this is particularly sane. Ideas?

Reply:
1. I'll check if inlining it's appropriate, at the very least I agree that the `get` is misleading
2. I'm not sure I'm following, what do you mean by "not having `pipeline-description.yml`?

Reply:
2. If `pipeline-description.yml` is not there, we simply do nothing. It seems to me that leaves `CLIProject` in a sort of a zombie state. Can it actually be used without `pipeline-description.yml`?
   Maybe we need `revng dump-pipeline-description` *now*.

Reply:
Let's make things more straightforward. Let's require that we have whatever is needed at construction time (if necessary let's find a way to obtain `pipeline-description.yml`) on demand.

Reply:
So... `revng dump-pipeline-description`?

Reply:
I suggest `revng init --resume=` which just set up the `resume` directory, so that if we need more stuff (like, idk, the model) it's already there.

Otherwise, the simplest thing would to have an artifact in one of the empty steps of the pipeline which dumps the binary and then we do `revng artifact initial /dev/null --resume=...`.

Reply:
Added `revng init`